platform/android/java/build.gradle

```diff
@@ -138,6 +143,10 @@ def templateBuildTasks() {
 
     // Only build the apks and aar files for which we have native shared libraries.
     for (String target : supportedTargets) {
+        if (target == "release_debug") {
+            // 'release_debug' is not supported for generating templates.
+            continue
+        }
```

Comment:
There seems to be a inconsistency between the Gradle and SCons targets.

"Debug" template *are* built with `scons tools=no target=release_debug`. The `target=debug` platform is only used for actual engine debugging (i.e. getting dev asserts, more information on crashes, etc.), and we don't ship any binaries built with `target=debug` (this is only for custom builds for people who know they need it).

I see that `platform/android/detect.py` doesn't seem to do anything special for `target=release_debug` compared to `target=release`, which is a discrepancy with all other platforms (note that it's `startswith`, applies to both `release` and `release_debug`): https://github.com/godotengine/godot/blob/f5b9cbaff6f0a058d08187b3124948ae68848cd0/platform/android/detect.py#L189

Reply:
> I see that `platform/android/detect.py` doesn't seem to do anything special for `target=release_debug` compared to `target=release`, which is a discrepancy with all other platforms (note that it's `startswith`, applies to both `release` and `release_debug`):

Actually that part seems to be intentional, the difference is made in `SConstruct` directly for all platforms, seems like we factored this out at some point. Other platforms like Linux still make a difference between the two in their `detect.py` but it's just to switch from `-O3` to `-O2` between `release` and `release_debug` (and on Android we seem to only use `-O2` so it's not needed for now).

Reply:
@akien-mga I have pushed a second (separate) commit that attempts to resolve the inconsistency by having the gradle's `buildtypes` match exactly the scons' targets.

This has the side effect of affecting the naming of the generated template apks (`android_debug.apk` is now `android_release_debug.apk`). I think that's a safe compromise to make since the template apks are not directly exposed to the end user.